libursa/src/errors/mod.rs

```diff
@@ -2,197 +2,217 @@
-            UrsaCryptoError::InvalidParam6(ref description) => write!(f, "Invalid param 4: {}", description),
-            UrsaCryptoError::InvalidParam7(ref description) => write!(f, "Invalid param 4: {}", description),
-            UrsaCryptoError::InvalidParam8(ref description) => write!(f, "Invalid param 4: {}", description),
-            UrsaCryptoError::InvalidParam9(ref description) => write!(f, "Invalid param 4: {}", description),
-            UrsaCryptoError::InvalidState(ref description) => write!(f, "Invalid library state: {}", description),
-            UrsaCryptoError::InvalidStructure(ref description) => write!(f, "Invalid structure: {}", description),
-            UrsaCryptoError::IOError(ref err) => err.fmt(f),
-            UrsaCryptoError::AnoncredsRevocationAccumulatorIsFull(ref description) => write!(f, "Revocation accumulator is full: {}", description),
-            UrsaCryptoError::AnoncredsInvalidRevocationAccumulatorIndex(ref description) => write!(f, "Invalid revocation accumulator index: {}", description),
-            UrsaCryptoError::AnoncredsCredentialRevoked(ref description) => write!(f, "Credential revoked: {}", description),
-            UrsaCryptoError::AnoncredsProofRejected(ref description) => write!(f, "Proof rejected: {}", description),
-        }
+impl From<log::SetLoggerError> for UrsaCryptoError {
+    fn from(err: log::SetLoggerError) -> UrsaCryptoError {
+        err.context(UrsaCryptoErrorKind::InvalidState).into()
     }
 }
 
-impl Error for UrsaCryptoError {
-    fn description(&self) -> &str {
-        match *self {
-            UrsaCryptoError::InvalidParam1(ref description) => description,
-            UrsaCryptoError::InvalidParam2(ref description) => description,
-            UrsaCryptoError::InvalidParam3(ref description) => description,
-            UrsaCryptoError::InvalidParam4(ref description) => description,
-            UrsaCryptoError::InvalidParam5(ref description) => description,
-            UrsaCryptoError::InvalidParam6(ref description) => description,
-            UrsaCryptoError::InvalidParam7(ref description) => description,
-            UrsaCryptoError::InvalidParam8(ref description) => description,
-            UrsaCryptoError::InvalidParam9(ref description) => description,
-            UrsaCryptoError::InvalidState(ref description) => description,
-            UrsaCryptoError::InvalidStructure(ref description) => description,
-            UrsaCryptoError::IOError(ref err) => err.description(),
-            UrsaCryptoError::AnoncredsRevocationAccumulatorIsFull(ref description) => description,
-            UrsaCryptoError::AnoncredsInvalidRevocationAccumulatorIndex(ref description) => description,
-            UrsaCryptoError::AnoncredsCredentialRevoked(ref description) => description,
-            UrsaCryptoError::AnoncredsProofRejected(ref description) => description,
+#[cfg(feature = "ffi")]
+impl From<UrsaCryptoErrorKind> for ErrorCode {
```

Comment:
Is this function intended to allow Indy to use Ursa without needing to change Indy? 
If so, is there a plan in the future to modify Indy to respond to Ursa's errors instead of requiring Ursa to do things like this?

Reply:
Not at the moment. But the plan is that ursa will follow its own development plan now and Indy will have to adapt to using ursa.